imap/jmap_backup.c

```diff
@@ -1726,15 +1714,15 @@ static void restore_mailbox_plan_cb(const char *guid __attribute__((unused)),
             if (i == 0) last_removed = rmail->removed;
 
             if (rmail->removed == last_removed) {
-                arrayu64_t *msgnos = hash_lookup(rmail->mboxname, mailboxes);
-                if (!msgnos) {
-                    /* Create msgno list for this mailbox */
-                    msgnos = arrayu64_new();
-                    hash_insert(rmail->mboxname, msgnos, mailboxes);
+                struct mailbox_plan *plan = hash_lookup(rmail->mboxname, mailboxes);
+                if (!plan) {
+                    /* Create a plan for this mailbox */
+                    plan = xzmalloc(sizeof(struct mailbox_plan));
+                    hash_insert(rmail->mboxname, plan, mailboxes);
                 }
 
-                /* Add this msgno to the mailbox */
-                arrayu64_append(msgnos, rmail->msgno);
+                /* Add this msgno to the restore array */
+                arrayu64_append(&plan->restore, rmail->msgno);
```

Comment:
Which raises a naming convention issue here that I didn't pick up before.  What's `msgno`?  The naming conventions say that `msgno` is the IMAP message number in a selected mailbox, i.e. an `index_state` offset.